deckhouse-controller/pkg/controller/deckhouse-release/cleanup_deckhouse_release.go

```diff
@@ -14,66 +14,57 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package hooks
+package deckhouse_release
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
 	"sort"
 	"time"
 
-	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
-	"github.com/flant/addon-operator/sdk"
-	"github.com/flant/shell-operator/pkg/kube/object_patch"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/utils/pointer"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/deckhouse/deckhouse/deckhouse-controller/pkg/apis/deckhouse.io/v1alpha1"
+	d8updater "github.com/deckhouse/deckhouse/deckhouse-controller/pkg/controller/deckhouse-release/updater"
 	"github.com/deckhouse/deckhouse/go_lib/updater"
-	d8updater "github.com/deckhouse/deckhouse/modules/002-deckhouse/hooks/internal/updater"
 )
 
-/*
-  This hook handles invalid situation when more then 1 Deployed release exists at the moment:
-    Hook moves all releases except the latest one to the Skipped state
-
-  The hook will keep only 10 Outdated releases, removing others
-*/
+func (r *deckhouseReleaseReconciler) cleanupDeckhouseReleaseLoop(ctx context.Context) {
+	for {
+		err := r.cleanupDeckhouseRelease(ctx)
+		if err != nil {
+			r.logger.Errorf("check Deckhouse release: %s", err)
+		}
 
-var _ = sdk.RegisterFunc(&go_hook.HookConfig{
-	Queue: "/modules/deckhouse/cleanup_deckhouse_release",
-	Kubernetes: []go_hook.KubernetesConfig{
-		{
-			Name:                         "releases",
-			ApiVersion:                   "deckhouse.io/v1alpha1",
-			Kind:                         "DeckhouseRelease",
-			ExecuteHookOnEvents:          pointer.Bool(false),
-			ExecuteHookOnSynchronization: pointer.Bool(true),
-			FilterFunc:                   filterDeckhouseRelease,
-		},
-	},
-}, cleanupReleases)
-
-func cleanupReleases(input *go_hook.HookInput) error {
-	snap := input.Snapshots["releases"]
-	if len(snap) == 0 {
-		return nil
+		time.Sleep(24 * time.Hour)
 	}
+}
 
-	now := time.Now().UTC()
+func (r *deckhouseReleaseReconciler) cleanupDeckhouseRelease(ctx context.Context) error {
+	var releases v1alpha1.DeckhouseReleaseList
+	err := r.client.List(ctx, &releases)
+	if err != nil {
+		return fmt.Errorf("get deckhouse releases: %w", err)
+	}
 
-	releases := make([]*d8updater.DeckhouseRelease, 0, len(snap))
-	for _, sn := range snap {
-		releases = append(releases, sn.(*d8updater.DeckhouseRelease))
+	pointerReleases := make([]*v1alpha1.DeckhouseRelease, 0, len(releases.Items))
+	for _, r := range releases.Items {
```

Comment:
NIT: shadowing. Reconciler has the same variable

Reply:
Fixed